lib/services/api/api_service.ts

```diff
@@ -234,4 +256,56 @@ export class SNApiService extends PureService {
 
     return response;
   }
+
+  private async checkForExpiredAccessToken(errorResponse: HttpResponse) {
+    if (errorResponse.status !== EXPIRED_ACCESS_TOKEN_RESPONSE_STATUS) {
+      return;
+    }
+    await this.refreshSession();
+  }
+
+  private async refreshSession() {
+    if (this.refreshingSession) {
+      return;
+    }
+    const user = await this.storageService!.getValue(StorageKey.User);
+    if (!user) {
```

Comment:
Why would refreshSession be called if user is undefined? It seems like we would never get to this point. This sounds like it should throw an exception instead.

```
if(!user || !user.uuid) {
   throw...
}
```

But again, will refreshSession ever be called in an offline context? If not, this check doesn't need to exist.